src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,374 @@
+    public static final String SPACE = " ";
+    public static final String DELIMITER_DEADLINE = " /by ";
+    public static final String DELIMITER_EVENT = " /at ";
+    public static final String MESSAGE_NULL_COMMAND_ARGS_A = "☹ OOPS!!! The description of a ";
+    public static final String NULL_COMMAND_ARGS_B = " cannot be empty.";
+    public static final String MESSAGE_NULL_INDEX_DONE_A = "☹ OOPS!!! The index of the task to be marked as done";
+    public static final String MESSAGE_NULL_INDEX_DONE_B = "cannot be empty.";
+    public static final String MESSAGE_NULL_INDEX_DELETE_A = "☹ OOPS!!! The index of the task to be deleted cannot be";
+    public static final String MESSAGE_NULL_INDEX_DELETE_B = "empty.";
+    public static final String MESSAGE_INVALID_COMMAND = "☹ OOPS!!! I'm sorry, but I don't know what that means :-(";
+    public static final String MESSAGE_DELETE_TASK = "Noted. I've removed this task:";
+    public static final String FILE_PATH = "data/duke.txt";
+    public static final String LETTER_TODO = "T";
+    public static final String LETTER_DEADLINE = "D";
+    public static final String LETTER_EVENT = "E";
+    public static final String MESSAGE_SOMETHING_WENT_WRONG = "Something went wrong: ";
+    private static ArrayList<Task> tasks;
+    private static boolean isExit;
+
+    public static void main(String[] args) {
+        showWelcomeMessage();
+        initProgram();
+        while (!isExit) {
+            String userInput = getUserInput();
+            try {
+                executeCommand(userInput);
+            }
+            catch (InvalidCommandException e) {
+                showInvalidCommandExceptionMessage();
+            }
+        }
+    }
+
+    private static void initProgram() {
+        tasks = new ArrayList<>();
+        isExit = false;
+        readTasks();
+    }
+
+    private static void readTasks() {
+        File f = new File(FILE_PATH);
+        try {
+            if (f.exists()) {
+                readFromFile();
+            } else {
+                f.createNewFile();
+            }
+        }
+        catch (IOException e) {
+            System.out.println(MESSAGE_SOMETHING_WENT_WRONG + e.getMessage());
+            isExit = true;
+        }
+    }
+
+    private static void readFromFile() throws FileNotFoundException {
+        File f = new File(FILE_PATH);
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String taskLine = s.nextLine();
+            String[] taskTypeAndParams = taskLine.split(" \\| ");
+            switch (taskTypeAndParams[0]) {
+            case LETTER_TODO:
+                tasks.add(new Todo(taskTypeAndParams[2]));
+                break;
+            case LETTER_DEADLINE:
+                tasks.add(new Deadline(taskTypeAndParams[2], taskTypeAndParams[3]));
+                break;
+            case LETTER_EVENT:
+                tasks.add(new Event(taskTypeAndParams[2], taskTypeAndParams[3]));
```

Comment:
You can consider using temporary variables as labels so that reviewers have a better understanding of how your command string is formatted.